### PR TITLE
feat: add get_source_assignments() function.

### PR DESCRIPTION
**Pull request by chrisdoehring**
**From @chrisdoehring** 
This is an update to allow reading Subject Source records (filtered by subject IDs or source IDs). It's first use is in an ER2ER sharing function that needs to lookup assigned-ranges for filtering observations as they're read from a source ER.

---
**From Copilot**

This pull request adds support for retrieving source assignments (also known as subject_sources) in the async client, along with comprehensive tests and several improvements to project configuration and code quality. The main changes include a new async method for fetching source assignments (with filtering), updates to the documentation, enhancements to testing infrastructure, and improvements to time zone handling in authentication logic.

**New feature: Async source assignments support**
- Added the `get_source_assignments` async method to `AsyncERClient`, allowing retrieval of source assignments with optional filtering by subject and source IDs.
- Added extensive tests for `get_source_assignments`, covering normal operation, filtering, and all major error and edge cases.
- Updated the README to document the new async client capability for getting source assignments.

**Testing and development workflow improvements**
- Added optional test dependencies and a test script to `pyproject.toml` for easier test setup and execution.
- Updated `.pre-commit-config.yaml` to exclude `erclient/__init__.py` from auto-formatters and linters, preventing unwanted formatting changes to this file.
- Updated `.isort.cfg` to skip `erclient/__init__.py` for import sorting.

**Code quality and maintenance**
- Improved timezone handling in authentication logic by replacing `pytz` usage with Python's standard library `datetime.timezone.utc`, simplifying code and removing a dependency. [[1]](diffhunk://#diff-72a66fbd8b127a5586d5324ed3a8c1affa269677777781f5f68af348b055b34cL8-R10) [[2]](diffhunk://#diff-72a66fbd8b127a5586d5324ed3a8c1affa269677777781f5f68af348b055b34cL97-R98) [[3]](diffhunk://#diff-72a66fbd8b127a5586d5324ed3a8c1affa269677777781f5f68af348b055b34cL136-R138) [[4]](diffhunk://#diff-72a66fbd8b127a5586d5324ed3a8c1affa269677777781f5f68af348b055b34cL1196-R1197) [[5]](diffhunk://#diff-72a66fbd8b127a5586d5324ed3a8c1affa269677777781f5f68af348b055b34cL1243-R1245)
- Updated version to `1.8.0` to reflect the new feature addition.
- Fixed and clarified exports in `erclient/__init__.py` to include new and existing exception classes.

### CODE REVIEW

File: erclient/client.py [hunk 1]
```diff
@@ -1335,6 +1336,19 @@ async def get_subject_sources(self, subject_id):
...
+        Get the source assignments (aka subject_sources). Optionally filter by subject_ids or source_ids.
+        """
+        params = {}
+        if subject_ids:
+            params['subjects'] = ','.join(subject_ids)
+
+        if source_ids:
+            params['sources'] = ','.join(source_ids)
+
+        return await self._get(f'subjectsources', params=params or None)
```
[the remark this answers is not in the document]
Comment (chrisdoehring) [reply erclient/client.py]: resolved locally.
